src/PoolManager.sol

```diff
@@ -203,34 +214,62 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         PoolId id = key.toId();
         _checkPoolInitialized(id);
 
-        key.hooks.beforeSwap(key, params, hookData);
-
-        uint256 feeForProtocol;
-        uint24 swapFee;
-        Pool.SwapState memory state;
-        (delta, feeForProtocol, swapFee, state) = pools[id].swap(
-            Pool.SwapParams({
-                tickSpacing: key.tickSpacing,
-                zeroForOne: params.zeroForOne,
-                amountSpecified: params.amountSpecified,
-                sqrtPriceLimitX96: params.sqrtPriceLimitX96
-            })
-        );
-
-        _accountPoolBalanceDelta(key, delta);
-
-        // the fee is on the input currency
-        unchecked {
-            if (feeForProtocol > 0) {
-                protocolFeesAccrued[params.zeroForOne ? key.currency0 : key.currency1] += feeForProtocol;
+        int128 hookDeltaInSpecified = key.hooks.beforeSwap(key, params, hookData);
```

Comment:
ok so for the record, the initial Custom Accounting was all or nothing, you were either using the v3 curve for your modifyLiquidity, swap, donate, or you weren't. One flag. And it _does_ "no-op" (no afterSwap would get hit) Hook would have to preemptively take/settle/mint/burn in beforeSwap.

 without delving too much into nitty gritty impl, the original custom accounting idea we implemented would've looked something like this:

```
if (key.hooks.isCustomAccounting()) {
  delta = key.hooks.beforeSwap();
   _accountDeltaFor(key.currency0, -delta.amount0(), address(key.hooks))
   _accountDeltaFor(key.currency1, -delta.amount1(), address(key.hooks))
  return delta;
} else {
  key.hooks.beforeSwap();
}

....rest of swap function stays the same
```